src/pages/payments/refund/Refund.tsx

```diff
@@ -0,0 +1,213 @@
+ */
+
+import { Card, Element } from '@invoiceninja/cards';
+import { Button, InputField, SelectField } from '@invoiceninja/forms';
+import axios, { AxiosError } from 'axios';
+import { endpoint } from 'common/helpers';
+import { Invoice } from 'common/interfaces/invoice';
+import { ValidationBag } from 'common/interfaces/validation-bag';
+import { defaultHeaders } from 'common/queries/common/headers';
+import { usePaymentQuery } from 'common/queries/payments';
+import { Alert } from 'components/Alert';
+import Toggle from 'components/forms/Toggle';
+import { useFormik } from 'formik';
+import { useEffect, useState } from 'react';
+import toast from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+import { useQueryClient } from 'react-query';
+import { generatePath, useParams } from 'react-router-dom';
+
+export function Refund() {
+  const queryClient = useQueryClient();
+  const { id } = useParams();
+  const [t] = useTranslation();
+  const { data: payment } = usePaymentQuery({ id });
+  const [errors, setErrors] = useState<ValidationBag>();
+  const [invoices, setInvoices] = useState<string[]>([]);
+  const [email, setEmail] = useState(false);
+  const formik = useFormik({
+    enableReinitialize: true,
+    initialValues: {
+      id: payment?.data.data.id,
+      date: payment?.data.data.date,
+
+      invoices: [],
+    },
+    onSubmit: (values) => {
+      const toastId = toast.loading(t('processing'));
+      setErrors(undefined);
+      axios
+        .post(
+          endpoint('/api/v1/payments/refund?&email_receipt=:email', { email }),
+          values,
+          {
+            headers: defaultHeaders,
+          }
+        )
+        .then(() => {
+          toast.success(t('updated_payment'), { id: toastId });
+        })
+        .catch((error: AxiosError) => {
+          console.error(error);
+          toast.error(t('error_title'), { id: toastId });
+          if (error.response?.status === 422) {
+            setErrors(error.response.data);
+          }
+        })
+        .finally(() => {
+          formik.setSubmitting(false);
+          queryClient.invalidateQueries(
+            generatePath(`/api/v1/payments/refund?&email_receipt=${email}`)
+          );
+        });
+    },
+  });
+
+  useEffect(() => {
+    invoices.map((invoiceId: string) => {
+      const invoiceItem = payment?.data.data.invoices.find(
+        (invoice: Invoice) => invoice.id == invoiceId
+      );
+      if (invoiceItem)
+        formik.setFieldValue('invoices', [
+          ...formik.values.invoices,
+          {
+            amount:
+              invoiceItem?.paid_to_date >
+              payment?.data.data.amount - payment?.data.data.refunded
+                ? payment?.data.data.amount - payment?.data.data.refunded
+                : invoiceItem?.paid_to_date,
+            invoice_id: invoiceItem?.id,
+            credit_id: '',
+            id: '',
+          },
```

Comment:
Is it possible to extract this into a separate method with a nice name so we know what's happening?